### PR TITLE
fix(ext/node): make worker setup synchronous

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit fixes race condition in "node:worker_threads" module were
the first message did a setup of "threadId", "workerData" and "environmentData".
Now this data is passed explicitly during workers creation and is set up
before any user code is executed.

Closes https://github.com/denoland/deno/issues/22783
Closes https://github.com/denoland/deno/issues/22672

### CODE REVIEW

File: ext/node/polyfills/worker_threads.ts [hunk 1]
```diff
@@ -192,6 +192,12 @@ class NodeWorker extends EventEmitter {
       name = "[worker eval]";
     }
     this.#name = name;
+
+    const maybeWorkerData = options?.workerData;
+    const serializedWorkerData = maybeWorkerData
+      ? core.serialize(maybeWorkerData)
+      : undefined;
```
Comment (bartlomieju) [inline ext/node/polyfills/worker_threads.ts:199]: This is wrong - we should use `serializeJsMessageData` here; and most likely include `environmentData` with `options.transferList` too.

File: runtime/js/99_main.js [hunk 2]
```diff
@@ -908,8 +909,18 @@ function bootstrapWorkerRuntime(
   // existing global `Deno` with `Deno` namespace from "./deno.ts".
   ObjectDefineProperty(globalThis, "Deno", core.propReadOnly(finalDenoNs));
 
+  let workerData = undefined;
+  if (maybeWorkerData) {
+    workerData = core.deserialize(maybeWorkerData);
```
Comment (bartlomieju) [inline runtime/js/99_main.js:914]: Wrong - should use `deserializeJsMessageData`

File: runtime/web_worker.rs [hunk 3]
```diff
@@ -616,14 +621,29 @@ impl WebWorker {
...
       let undefined = v8::undefined(scope);
+      let mut worker_data: v8::Local<v8::Value> = v8::undefined(scope).into();
+      if let Some(buf) = self.maybe_worker_data.take() {
+        let len = buf.len();
+        let store = v8::ArrayBuffer::new_backing_store_from_boxed_slice(
+          buf.into_boxed_slice(),
+        );
+        let ab =
+          v8::ArrayBuffer::with_backing_store(scope, &store.make_shared());
+        let v8_buf = v8::Uint8Array::new(scope, ab, 0, len).unwrap();
```
Comment (bartlomieju) [inline runtime/web_worker.rs:632]: I think it's okay to unwrap here.

**Review by mmastrac — COMMENTED** [review]
Much better. Do we have a test for the worker data being a shared array buffer?

**Review by mmastrac — APPROVED** [review]
LGTM with follow-up PR planned

**Pull request merged**